src/credentials/GroupcastDataProvider.cpp

```diff
@@ -0,0 +1,245 @@
+/*
```

Comment:
Shouldn't the GroupcastDataProvider be part of the cluster implementation in
src/app/clusters/groupcast/ and be injectable in the GroupCastLogic class  at the cluster instanciation rather than in server.ccp

Reply:
Since only GroupCastLogic use it, it is tempting to implement all the provider there. However:
1. The provider is the storage implementation. It make sense to keep the storage apart, since it could be implemented in may different ways, regardless of the cluster logic.
2. The new provider will be an extension of the existing GroupDataProvider. Makes sense to keep the structure consistent.
3. I haven't found a way to initialize the GroupCastLogic with the injected PersistentStorageDelegate.